sdk/search/Azure.Search.Documents/src/AzureSearchDocumentsEventSource.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Diagnostics.Tracing;
+using Azure.Core.Diagnostics;
+
+namespace Azure.Search.Documents
+{
+    [EventSource(Name = EventSourceName)]
+    internal sealed class AzureSearchDocumentsEventSource : EventSource
+    {
+        private const string EventSourceName = "Azure-Search-Documents";
+
+        internal const int BatchActionCountUpdatedEvent = 1;
+        internal const int BatchSubmittedEvent = 2;
+        internal const int RetryQueueResizedEvent = 3;
+        internal const int PendingQueueResizedEvent = 4;
+
+        private AzureSearchDocumentsEventSource() :
+            base(EventSourceName, EventSourceSettings.Default, AzureEventSourceListener.TraitName, AzureEventSourceListener.TraitValue)
+        { }
+
+        public static AzureSearchDocumentsEventSource Instance { get; } = new AzureSearchDocumentsEventSource();
```

Comment:
Nit - I think we've typically been calling these `Shared`

Reply:
We are using [`Singleton`](https://adpnetsource.azurewebsites.net/Azure.Identity/R/98bb7a765a3c850a.html), [`Log`](https://adpnetsource.azurewebsites.net/#Azure.Messaging.EventHubs/Diagnostics/EventHubsEventSource.cs,c960e5896e4dd589,references) and [`Instance`](https://adpnetsource.azurewebsites.net/Azure.IoT.ModelsRepository/R/0bdbedfafbc63193.html) as the name for the exposed singleton instances in different places.
I prefer the term `Instance` over `Singleton`. `Log` is not quite semantically correct and it is not also a noun for the object.

If there is no strong preference, I would prefer to call these as `Instance`.